samples/nrf9160/memfault/prj.conf

```diff
@@ -39,7 +39,7 @@ CONFIG_MEMFAULT_NCS_PROJECT_KEY=""
 CONFIG_MEMFAULT_SHELL=y
 CONFIG_MEMFAULT_HTTP_PERIODIC_UPLOAD=y
 # Configure the option below to set the periodic data upload interval
-# CONFIG_MEMFAULT_HTTP_PERIODIC_UPLOAD_INTERVAL_SECS=3600
+# CONFIG_MEMFAULT_HTTP_PERIODIC_UPLOAD_INTERVAL_SECS=1800
```

Comment:
Why don't we have a dedicated fragment for this ?
That will allow our users to do:
```
west build ... -- -DOVERLAY_CONFIG=-overlay-short_upload_freq.conf
```
or similar, and not having to modify an existing prj.conf.

Reply:
I'm not sure I see the benefit of adding an overlay for this single option. Wouldn't that just require the user to change the overlay file if one wants to change the value instead of `prj.conf`? I'd rather remove it if it's a problem to have it in

Reply:
I chose to remove the commented-out option as it's already documented in the sample and library documentation as one of the relevant options to configure.

Reply:
> Wouldn't that just require the user to change the overlay file if one wants to change the value instead of prj.conf? 

yes, if user want to adjust to something else than 1800, for example 900.
But the advantage of a Kconfig fragment is that user can easily copy that fragment to a new one and adjust to his needs.
```
$ cp <fragment>.conf my_adjustments.conf
edit my_adjustments.conf
$ west build ... -- -DOVERLAY_CONFIG=my_adjustments.conf
```
and user will not have to modified files in the repo.
(User will of course have a new file in the repo, but he could choose to place that file elsewhere, like `<workspace>/custom/<sample>/my_adjustments.conf`)

But for just a single option, `-DCONFIG_MEMFAULT_HTTP_PERIODIC_UPLOAD_INTERVAL_SECS=1800` is just as good.

Reply:
Alright, thanks for explaning. Will keep that in mind next time around

Reply:
The reason for not having commented options inside the prj.conf, is that we direct the user in sub-optimal flow where they will need to open `prj.conf` files just to see what else we propose.

With extra Kconfig fragments, then we are highlighting extra possibilities with the sample.
For example:
```
$ ls *.conf
prj.conf
# Hmm, me wondering if I should open that file for potential other nice to know options.
```
compared to:
```
$ ls *.conf
prj.conf    overlay-extra-settings.conf
# Oh, there is an extra set of options available, maybe I should look at what this does.
```

So with the extra file, it is much more visible to the users that we provide an alternative set of conf for some settings, whereas hiding it inside `prj.conf` force users to look into those files to see if we have other proposals.